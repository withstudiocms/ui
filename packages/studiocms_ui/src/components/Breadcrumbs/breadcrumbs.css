.sui-breadcrumbs {
  display: flex;
  flex-direction: row;
  gap: .25rem;
}

.sui-breadcrumbs > *:last-child {
  display: none;
}

.sui-breadcrumbs a {
  color: hsl(var(--primary-base));
  text-decoration: none;
}

.sui-breadcrumbs a:hover {
  text-decoration: underline;
}
