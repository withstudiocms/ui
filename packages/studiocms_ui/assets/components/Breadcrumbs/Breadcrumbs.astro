---
import './breadcrumbs.css';

type Props = {
  segments: {
    label: string;
    segment: string;
  }[];
  separator?: string;
}

const { segments, separator = '>' } = Astro.props;

const breadcrumbs = segments.map(({ label }, index) => {
  const path = segments.slice(0, index + 1).map(({ segment }) => segment).join('/').replaceAll("//", "/");
  return {
    label,
    path
  };
});
---

<div class="sui-breadcrumbs">
  {breadcrumbs.map(({ label, path }) => (
    <a href={path || '/'}>{label}</a><span>{separator}</span>
  ))}  
</div>
