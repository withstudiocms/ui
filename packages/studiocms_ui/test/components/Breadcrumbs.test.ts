import type { ComponentProps } from 'astro/types';
import { describe, expect } from 'vitest';
import Breadcrumbs from '../../src/components/Breadcrumbs/Breadcrumbs.astro';
import { test } from '../fixtures/vitest/AstroContainer';

type BreadCrumbsProps = ComponentProps<typeof Breadcrumbs>;

const props: BreadCrumbsProps = {
	segments: [
		{ label: 'Home', segment: '/' },
		{ label: 'About', segment: 'about' },
		{ label: 'Contact', segment: 'contact' },
	],
};

describe('Breadcrumbs Component', () => {
	test('renders Breadcrumbs component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(Breadcrumbs, 'Breadcrumbs', { props });
		expect(result).toMatchSnapshot();
	});

	test('renders Breadcrumbs component with custom separator', async ({ renderComponent }) => {
		const result = await renderComponent(Breadcrumbs, 'Breadcrumbs', {
			props: { ...props, separator: '>' },
		});
		expect(result).toMatchSnapshot();
	});

	test('renders Breadcrumbs component with single segment', async ({ renderComponent }) => {
		const result = await renderComponent(Breadcrumbs, 'Breadcrumbs', {
			props: { segments: [{ label: 'Home', segment: '/' }] },
		});
		expect(result).toMatchSnapshot();
	});

	test('renders Breadcrumbs component with no segments', async ({ renderComponent }) => {
		const result = await renderComponent(Breadcrumbs, 'Breadcrumbs', { props: { segments: [] } });
		expect(result).toMatchSnapshot();
	});
});
