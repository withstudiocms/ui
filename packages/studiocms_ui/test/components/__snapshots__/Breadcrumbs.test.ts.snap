// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumbs Component > renders Breadcrumbs with props { segments: [ { label: 'Home', segment: '/' } ] } 1`] = `"<div class="sui-breadcrumbs"> <a href="/">Home</a><span>&gt;</span> </div>"`;

exports[`Breadcrumbs Component > renders Breadcrumbs with props { segments: [ { label: 'Home', segment: '/' }, { label: 'About', segment: 'about' }, { label: 'Contact', segment: 'contact' } ] } 1`] = `"<div class="sui-breadcrumbs"> <a href="/">Home</a><span>&gt;</span><a href="/about">About</a><span>&gt;</span><a href="/about/contact">Contact</a><span>&gt;</span> </div>"`;

exports[`Breadcrumbs Component > renders Breadcrumbs with props { segments: [] } 1`] = `"<div class="sui-breadcrumbs">  </div>"`;

exports[`Breadcrumbs Component > renders Breadcrumbs with separator "{ separator: '/', segments: [ { label: 'Home', segment: '/' }, { label: 'Products', segment: '/products' }, { label: 'Electronics', segment: '/products/electronics' } ] }" 1`] = `"<div class="sui-breadcrumbs"> <a href="/">Home</a><span>/</span><a href="//products">Products</a><span>/</span><a href="//products/products/electronics">Electronics</a><span>/</span> </div>"`;

exports[`Breadcrumbs Component > renders Breadcrumbs with separator "{ separator: '>', segments: [ { label: 'Home', segment: '/' }, { label: 'Products', segment: '/products' }, { label: 'Electronics', segment: '/products/electronics' } ] }" 1`] = `"<div class="sui-breadcrumbs"> <a href="/">Home</a><span>&gt;</span><a href="//products">Products</a><span>&gt;</span><a href="//products/products/electronics">Electronics</a><span>&gt;</span> </div>"`;

exports[`Breadcrumbs Component > renders Breadcrumbs with separator "{ separator: '|', segments: [ { label: 'Home', segment: '/' }, { label: 'Products', segment: '/products' }, { label: 'Electronics', segment: '/products/electronics' } ] }" 1`] = `"<div class="sui-breadcrumbs"> <a href="/">Home</a><span>|</span><a href="//products">Products</a><span>|</span><a href="//products/products/electronics">Electronics</a><span>|</span> </div>"`;

exports[`Breadcrumbs Component > renders Breadcrumbs with separator "{ separator: '»', segments: [ { label: 'Home', segment: '/' }, { label: 'Products', segment: '/products' }, { label: 'Electronics', segment: '/products/electronics' } ] }" 1`] = `"<div class="sui-breadcrumbs"> <a href="/">Home</a><span>»</span><a href="//products">Products</a><span>»</span><a href="//products/products/electronics">Electronics</a><span>»</span> </div>"`;

exports[`Breadcrumbs Component > renders Breadcrumbs with separator "{ separator: '›', segments: [ { label: 'Home', segment: '/' }, { label: 'Products', segment: '/products' }, { label: 'Electronics', segment: '/products/electronics' } ] }" 1`] = `"<div class="sui-breadcrumbs"> <a href="/">Home</a><span>›</span><a href="//products">Products</a><span>›</span><a href="//products/products/electronics">Electronics</a><span>›</span> </div>"`;
